File: PYTHON/1328_Break_a_Palindrome.py
# Author: Atharv Patil
# Date: 18.10.22

class Solution:
    def breakPalindrome(self, a: str) -> str:
        if len(a) == 1:
            return ""
        i = 0
        j = len(a)-1
        while i < j:
            if a[i] != "a":
                return a[:i]+"a"+a[i+1:]
            i += 1
            j -= 1
        return a[:-1]+"b"

File: C++/1328_Break a Palindrome.cpp
// Name: Jyotideep Acharjee
// Date: 18/10/2022

class Solution {
public:
    string breakPalindrome(string palindrome) {
        int n = palindrome.size();
        if(palindrome.size()<=1) return "";
        
        string s = palindrome;
        int f = 0;
        for(int i=0;i<n/2;i++){
            //replave non - a char with a :
            if(s[i] != 'a'){
                f = 1;
                s[i] = 'a';
                return s;
            }
        }
        
        //if there are all 'a's then replace last with 'b' :
        if(!f) s[n-1] = 'b';
        return s;
    }
};
